doc/api/esm.md

```diff
@@ -1153,22 +1176,12 @@ for looking up conditional mappings elsewhere or to modify the list when calling
 the default resolution logic.
 
 The [current set of Node.js default conditions][Conditional Exports] will always
-be in the `context.conditions` list passed to the hook. If the hook wants to
-ensure Node.js-compatible resolution logic, all items from this default
-condition list **must** be passed through to the `defaultResolve` function.
+be in the `context.conditions` list passed to the hook. When calling to a parent
+loader, Node.js-compatible resolution behavior is not guaranteed unless all
+default conditions are passed.
 
 ```js
-/**
- * @param {string} specifier
- * @param {object} context
- * @param {string} context.parentURL
- * @param {string[]} context.conditions
- * @param {function} defaultResolve
- * @returns {object} response
- * @returns {string} response.url
- */
```

Comment:
Can you leave these alone for now?

Reply:
We should at least use the proper type annotations that we enforce across the rest of the documentation, and I don't think it's appropriate to push maintaining duplicate annotations onto node collabs.